graph/cytoscape/cytoscape.go

```diff
@@ -31,20 +32,21 @@ type NodeData struct {
 	Parent string `json:"parent,omitempty"` // Compound Node parent ID
 
 	// App Fields (not required by Cytoscape)
-	Service            string `json:"service"`
-	Version            string `json:"version,omitempty"`
-	Rate               string `json:"rate,omitempty"`               // edge aggregate
-	Rate3xx            string `json:"rate3XX,omitempty"`            // edge aggregate
-	Rate4xx            string `json:"rate4XX,omitempty"`            // edge aggregate
-	Rate5xx            string `json:"rate5XX,omitempty"`            // edge aggregate
-	RateSelfInvoke     string `json:"rateSelfInvoke,omitempty"`     // rate of selfInvoke
-	HasCircuitBreaker  string `json:"hasCB,omitempty"`              // true | false
-	HasRouteRule       string `json:"hasRR,omitempty"`              // true | false
-	IsDead             string `json:"isDead,omitempty"`             // true (has no pods) | false
-	IsGroup            string `json:"isGroup,omitempty"`            // set to the grouping type, current values: [ 'version' ]
-	IsRoot             string `json:"isRoot,omitempty"`             // true | false
-	IsUnused           string `json:"isUnused,omitempty"`           // true | false
-	HasMissingSidecars bool   `json:"hasMissingSidecars,omitempty"` // true | false
+	Service            string         `json:"service"`
+	Version            string         `json:"version,omitempty"`
+	Rate               string         `json:"rate,omitempty"`               // edge aggregate
+	Rate3xx            string         `json:"rate3XX,omitempty"`            // edge aggregate
+	Rate4xx            string         `json:"rate4XX,omitempty"`            // edge aggregate
+	Rate5xx            string         `json:"rate5XX,omitempty"`            // edge aggregate
+	RateSelfInvoke     string         `json:"rateSelfInvoke,omitempty"`     // rate of selfInvoke
+	HasCircuitBreaker  string         `json:"hasCB,omitempty"`              // true | false
+	HasRouteRule       string         `json:"hasRR,omitempty"`              // true | false
+	IsDead             string         `json:"isDead,omitempty"`             // true (has no pods) | false
+	IsGroup            string         `json:"isGroup,omitempty"`            // set to the grouping type, current values: [ 'version' ]
+	IsRoot             string         `json:"isRoot,omitempty"`             // true | false
+	IsUnused           string         `json:"isUnused,omitempty"`           // true | false
+	HasMissingSidecars bool           `json:"hasMissingSidecars,omitempty"` // true | false
+	Health             *models.Health `json:"health,omitempty"`
```

Comment:
All the other fields are simple "string" or "bool" values - this now introduces *models.Health. How large is this? Will this cause the size of the JSON to increase a lot? Is this going to be a problem?

Reply:
Well it is true that this is a lot bigger blob of info than we usually add to the json.  But, we don't add this on every node, which is good for size but maybe more important for performance as the health check isn't speedy.   It will be appended only for nodes tagged as needing health.  For the overview that's all nodes but remember that this is likely only a small number of nodes.  For the service graph it's not done at all (yet).  Perhaps we'd want to add it for nodes with request errors or something like that.

p.s. @jotak You may be interested in what I have here as I play around with this health stuff.  See also https://github.com/kiali/kiali-ui/pull/406.